lib/web/desktop_playback.go

```diff
@@ -176,7 +176,13 @@ func (pp *playbackPlayer) streamSessionEvents(ctx context.Context, cancel contex
 			if evt == nil {
 				pp.log.Debug("reached end of playback")
 
-				if _, err := pp.ws.Write([]byte("end")); err != nil {
+				msg, err := utils.FastMarshal(struct {
+					Message string `json:"message"`
+				}{Message: "end"})
```

Comment:
Would it be easier for the frontend if this pretended to be more like an audit event, rather than something custom?

Reply:
Not really. It's admittedly a little bit awkward, but it's simple to distinguish between this and the `DesktopRecording` events. I don't foresee this being the case, but if we end up needing more custom messages passed from the backend to the frontend, I'll see if I can think up something better.